core/src/wallet/ripple/RippleLikeAccount.cpp

```diff
@@ -461,5 +461,28 @@ namespace ledger {
             });
         }
 
+        void RippleLikeAccount::isAddressActivated(const std::string &address,
+                                                   const std::shared_ptr<api::BoolCallback> &isActivated) {
+            isAddressActivated(address).callback(getContext(), isActivated);
+        }
+
+        Future<bool> RippleLikeAccount::isAddressActivated(const std::string &address) {
+            auto xrpAddress = RippleLikeAddress::parse(address, getWallet()->getCurrency());
+            if (!xrpAddress) {
+                return Future<bool>::successful(false);
+            }
+            auto self = getSelf();
+            return _explorer->getBaseReserve().flatMap<bool>(getContext(), [self, xrpAddress](const std::shared_ptr<BigInt> &reserve) -> Future<bool> {
+                return self->_explorer->getBalance(
+                        std::vector<RippleLikeKeychain::Address>{
+                                std::dynamic_pointer_cast<RippleLikeAddress>(xrpAddress)
+                        }).map<bool>(self->getContext(), [reserve](const std::shared_ptr<BigInt> &balance) -> bool {
+                    if (!reserve || !balance) {
```

Comment:
You can move the `!reserve` test before calling that future to prevent an I/O since you will not use the result.